crates/strategy/zksync/src/cheatcode/runner/mod.rs

```diff
@@ -825,6 +825,16 @@ impl CheatcodeInspectorStrategyExt for ZksyncCheatcodeInspectorStrategyRunner {
                     }
                 }
 
+                // We increase the max depth to account for the inspector calling initialize_interp
+                // for nested calls in evm. This is because the expected_revert max
+                // depth is set in this function. Given that we don't execute
+                // initialize_interp in zkEVM calls we need to account for it.
+                // For the expect_revert max depth check increasing it by one is sufficient.
+                if let Some(expected_revert) = &mut state.expected_revert {
+                    expected_revert.max_depth =
+                        std::cmp::max(ecx.journaled_state.depth() + 1, expected_revert.max_depth);
```

Comment:
I'm unsure if the comment is actually the correct explanation of the behavior. `initialize_interp` is called multiple times during a test execution and it will set the max depth each time to the journaled state depth, how does it affect with a normal contract call such as in the test?

Reply:
The [initialize_interp](https://github.com/matter-labs/foundry-zksync/blob/9faa8a122319099a29cd74744884bab3f2771bdd/crates/cheatcodes/src/inspector.rs#L1326) updates the depth of the max depth of the calls if an expect revert is expected. In the case where we go into an external call, we do it in the zksync context, so we never update the max_depth reached by that call, so [this](https://github.com/matter-labs/foundry-zksync/blob/9faa8a122319099a29cd74744884bab3f2771bdd/crates/cheatcodes/src/test/revert_handlers.rs#L164) condition is always throwing the error.